src/ApiResource/ExistenceChecker.php

```
@@ -24,43 +16,25 @@ class ExistenceChecker implements ReplaceableHttpClientInterface
 {
     use ReplaceableHttpClientTrait;
 
-    private $httpClient;
-
     public function __construct(
-        GenericHttpClient $httpClient,
-        private readonly SerializerInterface $serializer,
+        private GenericHttpClient $httpClient,
         private readonly MicroservicePool $microservices,
     ) {
-        $this->httpClient = $httpClient;
     }
 
-    /**
-     * @param list<string> $iris
-     *
-     * @return array<string, bool>
-     *
-     * @throws ExceptionInterface
-     */
-    public function getExistenceStatuses(string $microserviceName, array $iris): array
+    public function resourceExists(string $microserviceName, string $iri): bool
     {
-        if (empty($iris)) {
-            return [];
-        }
-
-        $microservice = $this->microservices->get($microserviceName);
-
         $response = $this->httpClient->request(
-            $microservice,
-            'POST',
-            sprintf('/%s_check_resource', $microserviceName),
-            new ApiResourceExistenceCheckerPayload($iris),
-            'application/json',
-            'json'
+            $this->microservices->get($microserviceName),
+            'GET',
+            $iri,
         );
 
-        /** @var ApiResourceExistenceCheckerView $checkedIris */
-        $checkedIris = $this->serializer->deserialize($response->getContent(), ApiResourceExistenceCheckerView::class, 'json');
-
-        return $checkedIris->existences;
+        return match ($response->getStatusCode()) {
+            200, 203 => true,
+            404 => false,
+            // make it throw
+            default => (bool) $response->getContent(true),
+        };
```

Let's return true if `>= 200 && < 300`. I'm not sure why we need to throw an exception.

---

The exception is to support the current behaviour of `ApiResourceExist` (see `skipOnError`)